service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/FutureTaskResultToDomainObjectPromiseWrapper.java

```diff
@@ -28,40 +29,60 @@
 
 /**
  * Created by mohan.pandian on 18/03/15.
- *
+ * <p/>
  * A Promise that wraps a Future <TaskResult>, typically returned by TaskContext.executeAsyncCommand()
  * and eventually delivers the data from TaskResult (DomainObject)
  */
 public class FutureTaskResultToDomainObjectPromiseWrapper<DomainObject> implements Promise<DomainObject> {
 
-    private final Future<TaskResult> future;
-    private final List<PromiseListener> promiseListeners = new ArrayList<>();
+    private final List<Future<TaskResult>> futureList = new ArrayList<>();
     private PromiseBrokenException promiseBrokenException;
+    private ResponseMerger<DomainObject> responseMerger;
 
     public FutureTaskResultToDomainObjectPromiseWrapper(Future<TaskResult> future) {
-        this.future = future;
+        futureList.add(future);
+    }
+
+    public FutureTaskResultToDomainObjectPromiseWrapper(ResponseMerger<DomainObject> responseMerger) {
+        this.responseMerger = responseMerger;
     }
 
     @Override
     public boolean isRealized() {
-        return future.isDone();
+        for (Future<TaskResult> future : futureList) {
+            if (!future.isDone()) {
+                return false;
+            }
+        }
+        return true;
     }
 
     @Override
     public boolean isFullfilled() throws IllegalStateException {
-        return !future.isCancelled();
+        for (Future<TaskResult> future : futureList) {
+            if (future.isCancelled()) {
+                return false;
+            }
+        }
+        return true;
     }
 
     @Override
     public boolean isBroken() throws IllegalStateException {
-        return future.isCancelled();
+        for (Future<TaskResult> future : futureList) {
+            if (future.isCancelled()) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Isn't that we've to return true if future is cancelled and false otherwise. Looks wrong when compared to old code :)